app/src/main/java/com/github/swent/echo/compose/authentication/AuthenticationForm.kt

```diff
@@ -28,47 +31,133 @@ import com.github.swent.echo.R
  * @param action The text to be displayed on the action button.
  * @param onAuthenticate The callback to be invoked when the user clicks the action button. It
  *   receives the user's email and password as parameters.
+ * @param isOnline A boolean indicating whether the user is online.
+ * @param confirmPassword A boolean indicating whether the form should contain a field for
+ *   confirming the password.
+ * @param validate A boolean indicating whether the form should validate the input before calling
+ *   the [onAuthenticate] callback.
  */
 @Composable
 fun AuthenticationForm(
     action: String,
     onAuthenticate: (email: String, password: String) -> Unit,
     isOnline: Boolean,
+    confirmPassword: Boolean = false,
+    validate: Boolean = false,
 ) {
     var email by remember { mutableStateOf("") }
     var password by remember { mutableStateOf("") }
+    var confirm by remember { mutableStateOf("") }
+
+    var emailError by remember { mutableStateOf("") }
+    var passwordError by remember { mutableStateOf("") }
+    var confirmError by remember { mutableStateOf("") }
 
     Column(
         modifier = Modifier.fillMaxWidth(),
         horizontalAlignment = Alignment.CenterHorizontally,
     ) {
-        OutlinedTextField(
+        AuthenticationTextField(
             modifier = Modifier.fillMaxWidth().testTag("email-field"),
-            label = { Text(stringResource(R.string.authentication_form_email_label)) },
+            label = stringResource(R.string.authentication_form_email_label),
             value = email,
-            onValueChange = { email = it },
-            singleLine = true
+            onValueChange = {
+                emailError = ""
+                email = it
+            },
+            errorText = emailError,
         )
-        Spacer(modifier = Modifier.padding(16.dp))
-        OutlinedTextField(
+        Spacer(modifier = Modifier.padding(8.dp))
```

Comment:
magic number, it's better to specify 8.dp in a val at the beginning of the composable